src/charm.py

```diff
@@ -418,6 +446,52 @@ def _on_tls_certificate_pki_certificate_available(self, event: CertificateAvaila
         """Handle the tls-certificates-pki certificate available event."""
         self._add_ca_certificate_to_pki_secrets_engine()
 
+    def _on_vault_pki_certificate_creation_request(
+        self, event: CertificateCreationRequestEvent
+    ) -> None:
+        """Handle the vault-pki certificate creation request event."""
+        self._generate_pki_certificate_for_requirer(
```

Comment:
Wouldn't running this in the `_sync_vault_pki` be enough?

Reply:
Well that's what we do:

```python
def _sync_vault_pki(self) -> None:
        """Goes through all the vault-pki relations and sends necessary TLS certificate."""
        for relation in self.model.relations[PKI_RELATION_NAME]:
            outstanding_requests = self.vault_pki.get_outstanding_certificate_requests(
                relation_id=relation.id
            )
            for request in outstanding_requests:
                self._generate_pki_certificate_for_requirer(
                    csr=request.csr,
                    relation_id=relation.id,
                )
```

Reply:
I meant not to have a separate handler (`_on_vault_pki_certificate_creation_request`) for `certificate_creation_request`.
Couldn't we run `_configure`?

Reply:
I mean, we could but it's somewhat of an overkill. Plus we'd likely want to use the same logic for both PKI and KV